CppCoreGuidelines.md

```diff
@@ -19099,43 +19099,53 @@ and M functions each containing a `using namespace X`with N lines of code in tot
 
 ##### Note
 
-[Don't write `using namespace` in a header file](#Rs-using-directive).
+[Don't create namespace aliases in header files](#Rs-namespace-aliases).
 
 ##### Enforcement
 
 Flag multiple `using namespace` directives for different namespaces in a single source file.
 
-### <a name="Rs-using-directive"></a>SF.7: Don't write `using namespace` at global scope in a header file
+### <a name="Rs-namespace-aliases"></a>SF.7: Don't create namespace aliases in header files
 
 ##### Reason
 
-Doing so takes away an `#include`r's ability to effectively disambiguate and to use alternatives. It also makes `#include`d headers order-dependent as they may have different meaning when included in different orders.
+Header files should not create namespace aliases with `using namespace` or namespace declarations.
+Doing so takes away an `#include`r's ability to effectively disambiguate and to use alternatives. It also
+makes `#include`d headers order-dependent as they may have different meaning when included in different orders.
 
 ##### Example
 
-    // bad.h
+```cpp
+    // libx.h
     #include <iostream>
     using namespace std; // bad
 
+    namespace libx
+    {
+        using namespace liby; // bad
+    }
```

Comment:
There is a sound programming technique called **Namespace composition** documented in _The C++ Programming Language, 3rd edition_ which relies precisely on ability to have using declaratives insider another namespace in a header.  Furthermore, that ability is at the core of the _inline namespace_ facility which was introduced and is used for versioning of APIs in header files.